### PR TITLE
Support newer browsers not implementing addEventListener

### LINKED ISSUE #87: WebSocket.prototype.addEventListener is not a function in latest Chrome
An error is thrown at [line 50 in stream.js](https://github.com/maxogden/websocket-stream/blob/master/stream.js#L50). 
```
> WebSocket.prototype.addEventListener
undefined
```
Chrome version: `50.0.2661.94`
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) `addEventListener` is not a function on WebSocket, although it still works in Chrome Canary, Mozilla and Safari.

### PR DESCRIPTION
**Pull request by tothandras**
Fixes #87
Replace `addEventListener` calls with `onopen`, `onclose`, `onerror`, `onmessage`.

### CODE REVIEW

**Comment by mcollina** [comment]
why can't we just use `this.onopen()` and the like?

I'm 👎  on monkey patching the socket object.

**Comment by tothandras** [comment]
Alright, that's why I asked. :)
And what about the changes?

File: stream.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ function WebSocketStream(target, protocols, options) {
   var socketWrite = isBrowser ? socketWriteBrowser : socketWriteNode
   var proxy = through.obj(socketWrite, socketEnd)
 
-  if (protocols && !Array.isArray(protocols) && 'object' === typeof protocols) {
+  if (protocols && !Array.isArray(protocols) && typeof protocols === 'object') {
```
Comment (mcollina) [inline stream.js]: please don't invert this, it's not part of this change.

File: stream.js [hunk 2]
```diff
@@ -43,18 +43,28 @@ function WebSocketStream(target, protocols, options) {
   }
 
   // was already open when passed in
-  if (socket.readyState === 1) {
+  if (socket.readyState === WS.OPEN) {
```
Comment (mcollina) [inline stream.js]: Is `WS.OPEN` available everywhere?

**Comment by mcollina** [comment]
Changed are ok. However I will prefer to have a single way to set the events. Is that possible? Why can't we always use `onopen`  and the like?

File: stream.js (hunk 2 above)
Comment (tothandras) [inline stream.js]: I've checked it's part of the 'ws' library and the browsers too. But I can revert it if you would like to.

Comment (mcollina) [inline stream.js]: Ok for me.

**Comment by tothandras** [comment]
I don't see `onopen` and the like here: https://github.com/websockets/ws/blob/master/lib/WebSocket.js. :/

**Comment by tothandras** [comment]
Anything else? @mcollina

**Comment by mcollina** [comment]
Let's wait @mafintosh and @maxogden :)
